ocaml/xapi/xapi_vdi_helpers.ml

```diff
@@ -19,10 +19,26 @@ open Client
 open Db_cache_types
 open Pervasiveext
 open Threadext
+open Db_filter_types
 
 module D=Debug.Make(struct let name="xapi" end)
 open D
 
+let create_rrd_vdi ~__context ~sr =
+	let vdis = Db.VDI.get_refs_where ~__context ~expr:(And (
+		Eq (Field "SR", Literal (Ref.string_of sr)),
+		Eq (Field "type", Literal "rrd")
+	))
+	in
+	if vdis = [] then begin
```

Comment:
This could be simplified a bit with a pattern match, e.g.

```
match Db.VDI.get_refs_where ...
	...
	...
with
| [] -> begin
	...
	...
end
| vdi :: _ -> ...
```